app/Listeners/TaskStatusTimeCalculation.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+namespace App\Listeners;
+
+
+class TaskStatusTimeCalculation
+{
+    /**
+     * Handle the event
+     * @param \App\Events\TaskStatusTimeCalculation $event
+     */
+    public function handle(\App\Events\TaskStatusTimeCalculation $event)
+    {
+        $task = $event->model;
+
+        if ($task['collection'] === 'tasks' && $task->isDirty()) {
+            $updatedFields = $task->getDirty();
+            $unixTime = (new \DateTime())->format('U');
+
```

Comment:
1. `// If there's no owner, return false`
2. `// If owner exists, set $task->work = [...]` - read from dirty
3. `// Check for owner change`
...